src/frontend/hooks/server/invites.ts

```diff
@@ -17,16 +18,26 @@ export function usePendingInvites() {
   });
 }
 
-export function useAcceptInvite() {
+export function useAcceptInvite(projectId?: string) {
   const mapeoApi = useApi();
   const queryClient = useQueryClient();
+  const switchActiveProject = useUpdateActiveProjectId();
+
   return useMutation({
     mutationFn: async ({inviteId}: {inviteId: string}) => {
       if (!inviteId) return;
       mapeoApi.invite.accept({inviteId});
     },
     onSuccess: () => {
-      queryClient.invalidateQueries({queryKey: [INVITE_KEY]});
+      setTimeout(() => {
+        queryClient
+          .invalidateQueries({queryKey: [INVITE_KEY, PROJECTS_KEY]})
+          .then(() => {
+            if (projectId) {
+              switchActiveProject(projectId);
+            }
+          });
+      }, 5000);
```

Comment:
This setTimeout is a workaround.

Race condition is happening where:
1. invitee accepts invite
2. invitation resolves
3. when invitation resolves, the cache of `allprojects` is refetched
4. user is switched to the new project.

Between steps 2 and 3 there is a race condition, where the project has not been created yet, so the project in unavailable. This was discussed with @EvanHahn

Reply:
I could see adding a comment explaining this, but I hope to fix this soon in any case.